### ISSUE TITLE
Add support for HTTP/2

### ISSUE THREAD

**Issue by Brian3647** [Brian3647/snowboard#14]
See `ROADMAP.md` for more details.

**Comment by Brian3647** [comment]
http/2 is a way more complex implementation than http/1.1, so for security `h2` might be used. Opinions? or should the library use its own http/2 parsing the same way it parses http/1.1?

**Comment by neon-mmd** [comment]
> http/2 is a way more complex implementation than http/1.1, so for security `h2` might be used. Opinions? or should the library use its own http/2 parsing the same way it parses http/1.1?

Well h2 is good but I won't recommend it because when I run cargo audit on any crate using h2. It returns a whole bunch of security issues which has still not been patched. So I would suggest having your implementation. Also, having your implementation can also open up for more optimization which can drasticly improve the speed of the crate. :slightly_smiling_face:

**Comment by Brian3647** [comment]
that's a good point, but implementing the whole protocol from scratch will be a pain - maybe separate it into another crate?

**Comment by neon-mmd** [comment]
> that's a good point, but implementing the whole protocol from scratch will be a pain - maybe separate it into another crate?

Yes, but I think a better idea would be to have it as a seperate feature which will help both reduce code during compilation like if someone does not require http2 then they will just not enable it. So that way the code http2 will not be included. So I would suggest looking into the feature based conditional compilation approach. :slightly_smiling_face:

**Comment by Brian3647** [comment]
Sure, that's only a cfg statement
